pkg/platform/api/graphql/model/buildplanner/buildExpression.go

```diff
@@ -60,6 +60,7 @@ type BuildExpression struct {
 	expression       map[string]interface{}
 	solveNode        *map[string]interface{}
 	requirementsNode []interface{}
+	bytes            []byte
```

Comment:
bpModel.BuildExpression does have all the fields a BuildScript needs in order to convert from a bpModel.BuildExpression to a buildscript.BuildScript. Rather than extending BuildExpression, it's easier to work with its underlying JSON. If we make the incoming build expression's JSON available, we can construct a BuildScript from it.